test/jdk/com/sun/security/auth/module/AllPlatforms.java

```diff
@@ -21,44 +21,39 @@
  * questions.
  */
 
-/*
+ /*
  * @test
- * @bug 8039951
+ * @bug 8039951 8281717
  * @summary com.sun.security.auth.module missing classes on some platforms
  * @run main/othervm AllPlatforms
  */
-
 import javax.security.auth.login.Configuration;
 import javax.security.auth.login.LoginContext;
 import java.nio.file.Files;
 import java.nio.file.Paths;
 
 public class AllPlatforms {
+
+    private static final String OS = System.getProperty("os.name").toLowerCase();
+    private static final String UNIX_MODULE = "UnixLoginModule";
+    private static final String NT_MODULE = "NTLoginModule";
+
     public static void main(String[] args) throws Exception {
         login("cross-platform",
-                "UnixLoginModule", "optional",
-                "NTLoginModule", "optional",
-                "SolarisLoginModule", "optional");
-        try {
-            login("windows", "NTLoginModule", "required");
-            login("unix", "UnixLoginModule", "required");
```

Comment:
thanks for fixing this code. Test earlier skipped UnixLoginModule early on "nix" platform as windows login would fail.

Reply:
It will run on all platform even the unsuccessful ones appear before.